bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/RawType.java

```diff
@@ -18,36 +18,59 @@
  * Note that it is NOT adequate for any kind of streams, but only for fixed-size data.
  *
  * @author Kai Kreuzer
+ * @author Laurent Garnier - add MIME type
  *
  */
 public class RawType implements PrimitiveType, State {
 
+    private static final String DEFAULT_MIME_TYPE = "application/octet-stream";
+
     protected byte[] bytes;
+    protected String mimeType;
 
     public RawType() {
-        this(new byte[0]);
+        this(new byte[0], DEFAULT_MIME_TYPE);
     }
 
+    @Deprecated
     public RawType(byte[] bytes) {
+        this(bytes, DEFAULT_MIME_TYPE);
+    }
+
+    public RawType(byte[] bytes, String mimeType) {
         this.bytes = bytes;
+        this.mimeType = ((mimeType == null) || mimeType.isEmpty()) ? DEFAULT_MIME_TYPE : mimeType;
     }
 
     public byte[] getBytes() {
         return bytes;
     }
 
+    public String getMimeType() {
+        return mimeType;
+    }
+
     public static RawType valueOf(String value) {
-        return new RawType(Base64.getDecoder().decode(value));
+        int idx;
+        if (value == null || !value.startsWith("data:") || ((idx = value.indexOf(",")) < 0)) {
+            return new RawType();

```

Comment:
I would prefer to handle it similar to the point type and throw exception if invalid data is fit.